topo_descriptors/topo.py

```diff
@@ -128,10 +132,109 @@ def _tpi_kernel(size):
+        Corresponds to the diameter of the kernel used to compute it.
+    smth_factors (optional) : scalar or None or list with a combination of both.
+        Fraction(s) of the scale(s) at which the DEM is smoothed first (with a
+        gaussian filter). If None (default), no prior smoothing is performed.
+        If a scalar, the same fraction is used to determine the smoothing scale
+        of all specified scales. If a list, must match the length of arg 'scales'.
+    ind_nans (optional) : tuple of two 1D arrays
+        Contains the (row, column) indices of the NaNs in the original DEM to be
+        reassigned after computations. NaNs in the original DEM should be
+        interpolated prior computations as they propagate in convolutions with
+        the fast Fourier transform method (scipy.signal.convolve).
+    crop (optional) : dict
+        If specified the outputs are cropped to the given extend. Keys should be
+        the coordinates labels of dem_da and values should be slices of [min,max]
+        extend. Default is None.
+    outdir (optional) : string
+        The path to the output directory. Save to working directory by default.
+
+    See also
+    --------
+    std, circular_kernel
+    """
+
+    hlp.check_dem(dem_da)
+    logger.info(f"***Starting STD computation for scales {scales} meters***")
+    if not hasattr(scales, "__iter__"):
+        scales = [scales]
+    if not hasattr(smth_factors, "__iter__"):
+        smth_factors = [smth_factors] * len(scales)
+
+    scales_pxl, _ = hlp.scale_to_pixel(scales, dem_da)
+    sigmas = hlp.get_sigmas(smth_factors, scales_pxl)
+
+    for idx, scale_pxl in enumerate(scales_pxl):
+        logger.info(
+            f"Computing scale {scales[idx]} meters with smoothing factor"
+            f" {smth_factors[idx]} ..."
+        )
+        name = _std_name(scales[idx], smth_factors[idx])
+        array = std(dem=dem_da.values, size=scale_pxl, sigma=sigmas[idx])
+
+        array[ind_nans] = np.nan
+        hlp.to_netcdf(array, dem_da.coords, name, crop, outdir)
+        del array
+
+
+@hlp.timer
+def std(dem, size, sigma=None):
+    """Compute the standard deviation over a digital elevation model within
+    a rolling window.
+
+    Parameters
+    ----------
+    dem : array representing the DEM.
+    size : int
+        Size of the kernel for the convolution. Represents the diameter (i.e. scale)
+        in pixels at which the std is computed.
+    sigma (optional) : scalar
+        If provided, the DEM is first smoothed with a gaussian filter of standard
+        deviation sigma (in pixel size).
+
+    Returns
+    -------
+    array with local standard deviation values
+
+    See also
+    --------
+    scipy.signal.convolve, scipy.ndimage.gaussian_filter
+    """
+    kernel = circular_kernel(size)
+    kernel_sum = np.sum(kernel)
+    if sigma:
+        dem = ndimage.gaussian_filter(dem, sigma)
+
+    squared_dem = dem.astype("int32") ** 2
```

Comment:
why casting to int32 is needed ? Is it because the original DEM is in int16 ?

Reply:
yes, DEM can come in int16. In such cases, the maximum allowed value is 32,767, which is fine for elevation data in meters, but not enough for their squared value (ie cannot hold values larger than 32767^0.5 = 181 meters).